actions: notion fetcher 구현

bluelemon61 opened the pull request:
# Notion Fetcher 구현
- Github Action이 페이지에 필요한 데이터를 AID 노션에서 매일 자정에 불러도록 구현
- Github Action 자체로 scheduling mechanism을 내장하고 있어서 해당기능 사용

## 공유할 사항
- Github action이 Main 브런치에서만 작동하므로 머지 후 테스트 필요
- 현재 notion api key는 레포지토리 환경변수에 저장해 둔 상태

bluelemon61 committed:
actions: notion fetcher 구현

KimCookieYa commented on .github/workflows/schedule.yml:
```diff
@@ -0,0 +1,79 @@
+    - name: Run the fetch and update script
+      env:
+        NOTION_KEY: ${{ secrets.NOTION_KEY }}
+      run: |
+        node -e "
+        const fetch = require('node-fetch');
+        const { writeFile } = require('fs').promises;
+
+        async function fetchNotionData(pageID, notionKey) {
+          try {
+            const response = await fetch(\`https://api.notion.com/v1/databases/\${pageID}/query\`, {
+              method: 'POST',
+              headers: {
+                'Content-Type': 'application/json',
+                'Notion-Version': '2022-06-28',
+                Authorization: \`Bearer \${notionKey}\`
+              }
+            });
+            if (response.status !== 200) {
+              throw new Error(\`Failed to fetch data: \${response.status}\`);
+            }
+            return await response.json();
+          } catch (error) {
+            console.error(error);
+            return null;
+          }
+        }
+
+        (async () => {
+          const notionKey = process.env.NOTION_KEY;
+          const pageID = '28dfae8dfc694303861f61738dd50390';
+
+          const data = await fetchNotionData(pageID, notionKey);
+          if (!data) {
+            process.exit(1);
+          }
```
yml 파일 안에서 notion 데이터를 불러오는 api를 호출하는 로직을 담는 것은 좋지 않다고 생각함! 사실 기능 상에서나 성능 상에서나 문제는 없지만, cron 로직과 notion api 호출 로직은 다른 로직이니까 파일 단에서부터 분리해서 관리하는게 좋을 것 같어.

이런 경우,  로직을 별개의 .ts 파일로 분리하고, .yml 내에서는 해당 .ts 파일을 실행시키기만 하도록 변경해줘면 충분할 듯

KimCookieYa commented on .github/workflows/schedule.yml:
```diff
@@ -0,0 +1,79 @@
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # actions at midnight every day
```
github actions 가 돌아가는 서버 내에서의 시간은 한국 시간 기준이 아니라 UTC 기준이라 한국보다 9시간 빠름. 그래서 한국 시간으로 자정(0시)으로 하려면, 24-9 = 15시?
- cron: '0 15 * * *'로 하면 되지않을까

KimCookieYa reviewed (approved):
고생했으

bluelemon61 committed:
actions: 파일 분리

- ts 코드 실행 부분을 별도의 ts 파일로 분리
- 한국표준시 적용
- node-fetch 설치
  - 기본 node.js에 fetch 함수가 없고, nextjs 안에 있어서 따로 설치!

bluelemon61 commented:
- ts 코드 실행 부분을 별도의 ts 파일로 분리
- 한국표준시 적용
- node-fetch 설치
  - 기본 node.js에 fetch 함수가 없어서 따로 설치! (github action은 next.js 실행하지 않음)
